DeRhamCohomology/DifferentialForm.lean

```diff
@@ -43,3 +99,34 @@ theorem ederiv_ederiv_apply (ω : Ω^n⟮E, F⟯) {x} (h : ContDiffAt ℝ 2 ω x
 
 theorem ederiv_ederiv (ω : Ω^n⟮E, F⟯) (h : ContDiff ℝ 2 ω) : ederiv (ederiv ω) = 0 :=
   funext fun _ ↦ ederiv_ederiv_apply ω h.contDiffAt
+
+/- Pullback of a differential form -/
+def pullback (f : E → F) (ω : Ω^k⟮F, G⟯) : Ω^k⟮E, G⟯ :=
+  fun x ↦ (ω (f x)).compContinuousLinearMap (fderiv ℝ f x)
+
+theorem pullback_zero (f : E → F) :
+  pullback f (0 : Ω^k⟮F, G⟯) = 0 :=
+    rfl
+
+theorem pullback_add (f : E → F) (ω : Ω^k⟮F, G⟯) (τ : Ω^k⟮F, G⟯) :
+  pullback f (ω + τ) = pullback f ω + pullback f τ :=
+    rfl
+
+theorem pullback_sub (f : E → F) (ω : Ω^k⟮F, G⟯) (τ : Ω^k⟮F, G⟯) :
+  pullback f (ω - τ) = pullback f ω - pullback f τ :=
+    rfl
+
+theorem pullback_neg (f : E → F) (ω : Ω^k⟮F, G⟯) :
+  - pullback f ω = pullback f (-ω) :=
+    rfl
+
+theorem pullback_smul (f : E → F) (ω : Ω^k⟮F, G⟯) (c : ℝ) :
+  c • (pullback f ω) = pullback f (c • ω) :=
+    rfl
+
+theorem pullback_ofSubsingleton (f : E → F) (ω : F → F →L[ℝ] G) [Subsingleton (Fin k)] (i : Fin k) :
+  pullback f (ofSubsingleton i ω) = ofSubsingleton i (fun e ↦ (ω (f e)).comp (fderiv ℝ f e)) :=
+    rfl
+
+theorem pullback_constOfIsEmpty (f : E → F) [IsEmpty (Fin k)] (g : G) :
```

Comment:
`IsEmpty (Fin k)` is true only for `k = 0`, so I think that it's OK to state this for `k = 0` only.